chain/chain/src/chain.rs

```diff
@@ -1162,62 +1169,65 @@ impl Chain {
 
                 Ok(head)
             }
-            Err(e) => match e.kind() {
-                ErrorKind::Orphan => {
-                    let tail_height = self.store.tail()?;
-                    // we only add blocks that couldn't have been gc'ed to the orphan pool.
-                    if block.header().height() >= tail_height {
+            Err(e) => {
+                let _ = self.save_block_height_processed(block.header().height());
```

Comment:
We should only save the height if the block header signature matches the creator's, and the block content corresponds to what is in the headers.
It *seems* like right now it is always the case because the client does the check before calling downstream to `Chain::process_block`. If it is intended, we should either 
1. make it extremely clear that the chain interface is to only process blocks that have valid signatures (e.g. we need to explicitly remove the call to `check_validity` in `process_block` in `ChainUpdate`, and track down and remove the validation of the header, and either rename all the methods in `ChainUpdate` and `Chain` to something along the lines of `process_authenticated_block`, or add comments before each `process_block*` method on `Chain` and `ChainUpdate`), 
2. not make this assumption, and handle the case of a block with an invalid header or content in `Chain` properly (and have corresopnding tests, because end-2-end tests will naturally not catch it).

Reply:
I initially thought that 1 is the case, but then realized that we didn't actually process header when the provenance is SYNC, so going with 2 now.